commit eb69f594ec439309432e87834bead5276b7dbc9b
Author: Palmer Cox <p@lmercox.com>
Date:   Sun Feb 23 16:45:34 2025 -0500

    On Apple, use FIND_LIBRARY to locate libpq
    
    I think FIND_LIBRARY() is better than just relying on what pg_config
    said its libdir was, since, depending on how libpq was installed, it may
    or may not be in that directory. If its not, FIND_LIBRARY() is able to
    find it in other locations.

diff --git a/saga-gis/cmake/modules/FindPostgres.cmake b/saga-gis/cmake/modules/FindPostgres.cmake
index a4b6ec9ac..65e7ac69b 100644
--- a/cmake/modules/FindPostgres.cmake
+++ b/cmake/modules/FindPostgres.cmake
@@ -81,7 +81,7 @@ ELSE(WIN32)
         OUTPUT_VARIABLE PG_TMP
         OUTPUT_STRIP_TRAILING_WHITESPACE)
       IF (APPLE)
-        SET(POSTGRES_LIBRARY ${PG_TMP}/libpq.dylib CACHE STRING INTERNAL)
+        FIND_LIBRARY(POSTGRES_LIBRARY NAMES pq libpq PATHS ${PG_TMP})
       ELSEIF (CYGWIN)
         execute_process(COMMAND ${POSTGRES_CONFIG} --libs
           OUTPUT_VARIABLE PG_TMP
